index.bs

```diff
@@ -4241,19 +4242,26 @@ promises returned by its <a lt="writable stream writer">writer</a>'s {{WritableS
       },
 
       close() {
-        return new Promise((resolve, reject) => {
-          ws.onclose = resolve;
-          ws.close(1000);
-        });
+        return closeWS(1000);
       },
 
       abort(reason) {
-        return new Promise((resolve, reject) => {
-          ws.onclose = resolve;
-          ws.close(4000, reason && reason.message);
-        });
-      }
+        return closeWS(4000, reason && reason.message);
+      },
     });
+
+    function closeWS(code, reasonString) {
+      return new Promise((resolve, reject) => {
+        ws.onclose = e => {
```

Comment:
So "fail the WebSocket connection" points you to here https://tools.ietf.org/html/rfc6455#section-7.1.7 which tells you that the first thing we do it close the connection. Which then causes onclose to fire.

It might be that an implementation fires onerror much sooner than onclose, but Chrome doesn't do that. In fact, we tear down the connection synchronously when we detect a failure condition:

https://cs.chromium.org/chromium/src/net/websockets/websocket_channel.cc?q=%22Careful+study+of+RFC6455%22